### ISSUE TITLE
Export using `database` queue doesn't respect app locale

### ISSUE THREAD

**Issue by valpuia604** [laravel-filament/filament#12974]
### Package

filament/filament

### Package Version

v3.2.83

### Laravel Version

v11.8.0

### Livewire Version

v3.5

### PHP Version

v8.2

### Problem description

When switching to different locale and export data using `QUEUE_CONNECTION=database` doesn't respect app locale and always storing data in `en` language. if `QUEUE_CONNECTION=sync` then it's correct.

### Expected behavior

Export with queue respect app locale

### Steps to reproduce

clone, install
login using `test@example.com` and `password`
Change locale
Goto user, export user using non english locale
run queue `php artisan queue:work`
Check database/notifications

### Reproduction repository

https://github.com/valpuia604/fila-noti-db-bug

**Comment by zepfietje** [comment]
There's no way for the queue worker to know the selected locale when you're not storing it. Not a Filament bug, right? 🙂

**Comment by valpuia604** [comment]
I am not sure about "you're not storing it", how to store the locale? I hope we can do sth like this but don't know where exactly to change (for PR) https://laravel.com/docs/11.x/notifications#localizing-notifications

**Comment by zepfietje** [comment]
I'm not sure I fully understand the original issue. You're talking about data being stored in English, but we're talking about exports right? Not imports?

**Comment by valpuia604** [comment]
Yes, store in "English" only in export (I didn't check for imports atm). Like I said in OP using `sync` it's working fine.

Can you explain how we can store (not sure this word is correct or not) the selected locale for queue work `database`?

**Comment by zepfietje** [comment]
Could you clarify the issue here? I'm not sure what you're referring to with "storing in English".

**Comment by valpuia604** [comment]
![Screenshot from 2024-05-28 15-25-22](https://github.com/filamentphp/filament/assets/141896542/6d27dbae-babd-41a1-bc68-8dc3deb39f79)

As we can see from this image, my locale is `km` but after exported the title (Export completed), message (your league...) and download button text never translate because of queue `database`.. like I said these are working fine if I use `sync`

What I meant for "storing in English" is after exported, all these notifications are displayed based on database data, so my database contains only "English" word, not others (for `database` not `sync` queue). hope you understand

**Comment by zepfietje** [comment]
This should just work then, right?
https://laravel.com/docs/11.x/notifications#localizing-notifications

**Comment by valpuia604** [comment]
I hope that links resolves this, but I don't know how to apply that one in "Export"

**Comment by zepfietje** [comment]
Just implement the `HasLocalePreference` interface on your User model?

**Comment by Umar-Farooq-Shafi** [comment]
not working

```
    /**
     * @return string|null
     */
    public function preferredLocale(): ?string
    {
        return \App::getLocale();
    }
```

**Comment by chrimutz** [comment]
I have solved it this way.

In your Exporter class

```
public static function getCompletedNotificationTitle(Export $export): string
{
    $locale = $export->user()->getResults()->preferredLocale();

    return __(key:'filament-actions::export.notifications.completed.title', locale: $locale);
}
```

same at `getCompletedNotificationBody` method ..

**Comment by meetzoo** [comment]
In case you are using the language switcher from https://filamentphp.com/plugins/bezhansalleh-language-switch it stores the user language in a cookie or in the session, so if the export is not sync and is done on a separate queue, it won't know what the user language preference is.
I fixed it by adding the trait HasLocalePreference to the user model like mentioned [here](https://github.com/filamentphp/filament/issues/12974#issuecomment-2146674977) (localizing laravel notifications),  then adding the column 'locale' to the user table and saving the language to the db when the user switches the language.
Then in the user model I added so it takes the language from the DB

```
public function preferredLocale(): string{
        return ! empty($this?->locale) ? $this->locale : config('app.locale', 'en');
}
```
Another way would be to send current the language as a param when the export is initiated. Otherwise I don't know how the export job that is run on a separate queue would have access to the language the user had set.
